src/Guardrail.sol

```diff
@@ -222,18 +230,29 @@ contract Guardrail is ITransactionGuard, IModuleGuard, MultiSendCallOnlyv2 {
         bytes calldata,
         address
     ) external override {
-        bytes4 selector = _decodeSelector(data);
-        if (to == msg.sender && selector == GuardManager.setGuard.selector) {
+        _checkOperationAndAllowance(msg.sender, to, _decodeSelector(data), operation);
+    }
+
+    function _checkAfterExecution() internal {
+        address guard = abi.decode(SafeInterface(msg.sender).getStorageAt(uint256(GUARD_STORAGE_SLOT), 1), (address));
+        address moduleGuard =
+            abi.decode(SafeInterface(msg.sender).getStorageAt(uint256(MODULE_GUARD_STORAGE_SLOT), 1), (address));
```

Comment:
minor code style preference, since there's a lot happening on this line, you can reduce the cognitive load by:
1. extracting `SafeInterface(msg.sender)` to a separate variable and reusing it
2. can we store the storage slot as uint256 to avoid explicit conversion to uint256? or adjusting the interface